client/components/notice/index.tsx

```diff
@@ -0,0 +1,146 @@
+/**
+ * External dependencies
+ */
+
+import React, { useEffect, FunctionComponent, useRef } from 'react';
+import classnames from 'classnames';
+import { localize } from 'i18n-calypso';
+import Gridicon from 'calypso/components/gridicon';
+
+/**
+ * Internal dependencies
+ */
+import { TimerHandle } from 'calypso/types';
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+/**
+ * Module constants
+ */
+const GRIDICONS_WITH_DROP = [
+	'add',
+	'cross-circle',
+	'ellipsis-circle',
+	'help',
+	'info',
+	'notice',
+	'pause',
+	'play',
+	'spam',
+];
+
+/**
+ * Constants
+ */
+enum Status {
+	Error = 'is-error',
+	Info = 'is-info',
+	Success = 'is-success',
+	Warning = 'is-warning',
+	Plain = 'is-plain',
+}
+
+interface Props {
+	className?: string;
+	duration?: number;
+	onDismissClick: () => void;
```

Comment:
What I've observed during the refactoring away from Lodash's `noop` is that it tends to be cleaner if we don't provide a default prop to the function props, but then check if it's provided before using it. I think here it would be a good opportunity for refactoring, as if `onDismissClick` isn't set, we won't call a noop function unnecessarily, and all the related `setTimeout` / `clearTimeout` that are related to that.